rpc-test/tests/rpc.rs

```diff
@@ -228,7 +229,8 @@ fn test_rpc_slot_updates() {
     let rpc_pubsub_url = test_validator.rpc_pubsub_url();
 
     rt.spawn(async move {
-        let pubsub_client = PubsubClient::new(&rpc_pubsub_url).await.unwrap();
+        let request = rpc_pubsub_url.into_client_request().unwrap();
+        let pubsub_client = PubsubClient::new(request).await.unwrap();
```

Comment:
Changed this test so that we at least have one test that exercises the `Request`-as-input-argument codepath.